fnmatch/fnmatch.py

```diff
@@ -64,6 +79,29 @@ def fnmatchcase(name, pat):
     match = _compile_pattern(pat)
     return match(name) is not None
 
+_re_special_chars_map = {i: '\\' + chr(i) for i in b'()[]{}?*+-|^$\\.&~# \t\n\r\v\f'}
+def _re_translate(p, chmap):
+    oldp = p
+    for idx,c in enumerate(p):
+     print('[%d]=%s' % (idx, c))
+     if ord(c) in chmap:
+      newp = p.replace(c, chmap[ord(c)])
+      #print( '%s -> %s' % (oldp, newp) )
+      return newp
+    #print( '%s -> %s' % (oldp, p) )
+    return p
+
+def _re_escape(pattern):
```

Comment:
Would be best to add tests for this.

Reply:
I'm not familiar with tests; would need someone to assist on writing tests.

Reply:
See test_fnmatch.py in this directory